cache/search_stats.go

```diff
@@ -501,6 +501,32 @@ func (ssc *SearchStatsCacheImpl) loadSourcesByPositionAndParent() (map[string]st
 		key := fmt.Sprintf("%v-%v-%v", parent_uid, position, type_id)
 		ret[key] = source_uid
 	}
+
+	query = `select p.uid as parent_uid, gc.uid as source_uid, gc.type_id 
+	from sources p
+		join sources c on c.parent_id = p.id
+		join sources gc on gc.parent_id = c.id
+	where c.position is not null and p.name = 'TES'
+	order by c.position, gc.position`
+	rows, err = queries.Raw(ssc.mdb, query).Query()
+	if err != nil {
+		return nil, errors.Wrap(err, "queries.Raw")
+	}
+	defer rows.Close()
+	position := 0
+	for rows.Next() {
+		var parent_uid string // uid of parent source
+		var source_uid string // uid of child source
+		var type_id int64     // type of child source
+		position = position + 1
+		err = rows.Scan(&parent_uid, &source_uid, &type_id)
+		if err != nil {
+			return nil, errors.Wrap(err, "rows.Scan")
+		}
+		key := fmt.Sprintf("%v-%v-%v", parent_uid, position, type_id)
+		ret[key] = source_uid
```

Comment:
May this override an existing value? What could be the impact of the override for some searches?

Reply:
this won't override any search because key contains both - parent UID and division type ID and there was no at all previously search for TES using division type id 4, that is part.